### DISCUSSION TITLE
Telegram Notification Video/Picture URL

### DISCUSSION THREAD

**Discussion by svo08** [blakeblackshear/frigate#1876] [Ask A Question] [answered]
First thanks a lot for the project. Its really great.
I did manage to get to work with HA using external Frigate Instance with USB Coral including Integration.
I did manage to get a basic Telegram notification is working fine.  Means sending telegram message when person is detected on camara x only thing missing is the matching picture/video.

So i just need to add URL to send the correct Image (Camarea Name/ID, Type of dection "Person") with Telegram or even better matching Video.
Sadly i dont got the telegram notification code working which is shared here in documentaiton.

Can you please share the Picture/Video URL with me for person detection in camera x ?
Thank you very much.

Currently i have the follwing HA Automation Code.

'alias: Person in Frontdoor
description: ‘’
trigger:

    type: motion
    platform: device
    device_id: 68cce7cf5cf62f9fe3cd9c866335129c
    entity_id: binary_sensor.eingang_person_motion
    domain: binary_sensor
    condition: []
    action:
    service: notify.frigate_xxxxx_bot
    data:
    message: Detected a person on Frontdoor.
    mode: single’

**Comment by Minglarn** [comment]
Take advantage of MQTT ... Makes automation a lot easier... 

This is how my looks:
![image](https://user-images.githubusercontent.com/58695742/135645781-d0133340-7661-49a8-a6f5-b622f48075d2.png)
In the MQTT playload you've got a ton of usable info... 
```
service: telegram_bot.send_photo
data:
  url: >-
    http://192.168.1.97:5000/api/events/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg?bbox=0&crop=1
  caption: >-
    {{trigger.payload_json["after"]["camera"]|replace("frigate_","") | upper }}
    {{trigger.payload_json["after"]["label"] | upper }} 
    [VIEW VIDEO](https://top.seceret/api/events/{{trigger.payload_json["after"]["id"]}}/clip.mp4)
  target: 'xxxx'
  timeout: 1000
```
And my telegram spits out:
![image](https://user-images.githubusercontent.com/58695742/135651142-c2964aff-e6e3-493a-b234-55a2d41c60ef.png)

**Reply by svo08** [reply]
Hi Minglarn,

thx for you answer. At least i was able to use the first part.
So i got a snapshot.
But was not able to get work with Video.

So i did adjust the second url as well to my external frigate server or should it point to homeassistant correct?

http://external.frigate:5000/api/events/{{trigger.payload_json["after"]["id"]}}/clip.mp4

In my case i did use IPv4 Adress as well at least as long my cellpohone is on internal network it should work.
But i always got wrong url/file.

I would prefer to send it with telegram. But thats something later first i need a valid URL for the video.

**Reply by svo08** [reply]
Ok looks like i have to use Homeassistant URL.

So i did adjust it to HA Url
http://external.HA:8123/api/events/{{trigger.payload_json["after"]["id"]}}/clip.mp4

Now Mediaplayer does open when clicking on the link but sadly its still not playing.
Looks like its invalid marked play bottom ?

**Reply by svo08** [reply]
I did try both URLs without luck:
url: http://192.168.1.12:5000/api/events/{{trigger.payload_…
url: http://192.168.1.12:8123/api/frigate/notifications/ 2{{trig …

And API is enabled. I just added api: to configuration.yml

**Reply by JonGilmore** [reply]
So, just to be clear, you're supposed to point that URL at frigate (generally port 5000), not 8123. Also, just so it's understood, that URL won't resolve when you just hit it, you'll have to substitute in an event ID in order to make sure it works. You can get an event ID by downloading a program like MQTT Explorer to parse events as they come in.

**Reply by svo08** [reply]
Hi again thx for you help and sorry for that newbie questions.
Right, so my problem is to figure out correct API HTML String and figure out ID.

As i wrote in first post so i manged to put a trigger in place when a person is detected on Camera1  and do action
in automation via visual editor.
Then i did mange to attach a picture so i used http://10.0.0.42:5000/api/Camera1/person/best.jpg 
Which sends me Image of that last motion detection. Which is good.

You are right for that i need the ID somhow to build the string with MP4 Link.
Maybe you can help me again pointing to right scipt code.

Basically i like to do same. When im away i like to enable via Telegram a automation which sends me persons which are detected in backyard cam for example. I did get it to work with picuture and telegram and doing this but strugle to send video.

**Reply by JonGilmore** [reply]
The event ID comes out of the `{{trigger.payload_json['after']['id']}}` from the mqtt topic `frigate/events`. Because it's templated, it's dynamic.

**Reply by svo08** [reply]
Ok but do you have any clue why, URL for picture is working with event id but video URL not?
http://externalfrigate:5000/api/events/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg?bbox=0&crop=1
But Video URL below is not working in same yml?
http://externalfrigate:5000/api/events/{{trigger.payload_json["after"]["id"]}}/clip.mp4)
Maybe that Video URL is wrong but i did found the same and no other in HA Forum.
Even if i look for Video URL which only detects person on camera X.
But for start it would be nice to get one video URL working.

**Reply by JonGilmore** [reply]
Are you exposing port 5000 through your firewall? I'm using my internal IP and port within the automations. As I'd mentioned previously, you can download something like MQTT Explorer and find an example event id and test it out in your web browser.

**Reply by svo08** [reply]
Hi yes your are right i will again dig into the topic also play with MQTT Explorer it coudl be only that somehow the URL for Video Clip is wrong. Will try to sort it out.
No right now i dont plan to expose it, my thought was that sending it via Telegram via local API so i would not need in frist step direct access. I can use existing VPN and parallel NVR to check status in case of i have to doublecheck alram via telegram.
I did read some hints about secure remote frigate access but for now i prefer telegram. I have to look for some Hardware that can cover up to 6 streams.

**Comment by JonGilmore** [comment, marked as answer]
Here's maybe something that can help you out as well, I actually trim my clips to a shorter length using ffmpeg before sending them via telegram: https://github.com/JonGilmore/ha-personal/blob/05f7a55aac3554b78a2e58635fe92d372dbd5724/automation/frigate.yaml#L229-L262

**Reply by svo08** [reply]
Thx very much for sharing, looks like you doing exactly something what i will do.
Alarming when im out ... but i really have to take time and learn and review your code.

**Reply by JonGilmore** [reply]
let me know if I can help out in deciphering it! Some of the stuff is sorta specific to my situation, but happy to help!

**Reply by fhold** [reply]
wow thanks for sharing your automation, that is really helpful in optimizing my notifications with frigate
